Sources/Sentry/SentryTracer.m

```diff
@@ -344,7 +350,7 @@ - (void)setStartTimestamp:(nullable NSDate *)startTimestamp
 
 - (BOOL)isFinished
 {
-    return self.rootSpan.isFinished;
+    return self.rootSpan.isFinished && self.rootSpan.context.status != kSentrySpanStatusUndefined;
```

Comment:
h: Undefined is a valid value to finish a transaction.

```suggestion
    return self.rootSpan.isFinished;
```

Reply:
Given this was merged to main already in #2059, resolving.